packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -224,9 +237,45 @@ class DebugBundleFlutterAssets extends Target {
             resource.release();
           }
         }));
-    } catch (err, st){
+    } catch (err, st) {
       throw Exception('Failed to copy assets: $st');
     }
+    // Copy Info.plist template.
+    assetDirectory.parent.childFile('Info.plist')
+      ..createSync()
+      ..writeAsStringSync(r'''
```

Comment:
Why inline rather than a template file like iOS?

Reply:
I'm assuming that since this is ephemeral we don't want to require the user to manually create it each time.  Using the templating is complex, It's not tested as well as I like and it relies on mirrors via a 3rd party templating library.